File: apps/zLinebot/services/gpu-renderer/src/worker/worker.py
import time
from core.queue import dequeue
from core.render import render

def worker():

    while True:

        job = dequeue()

        if not job:
            time.sleep(3)
            continue

        try:
            render(job)
        except Exception as e:
            print(e)

if __name__ == "__main__":
    worker()

File: apps/zLinebot/services/tiktok-farm/src/workers/worker.js
import { dequeue } from "../core/queue.js"
import { getAccount } from "../core/accounts.js"
import { getProxy } from "../core/proxies.js"
import { upload } from "../uploader/uploader.js"

async function worker(){

while(true){

const job = await dequeue()

if(!job){

await new Promise(r=>setTimeout(r,5000))
continue

}

const account = await getAccount()
const proxy = await getProxy()

try{

await upload(account, proxy, job.video)

}catch(e){

console.error(e)

}

}

}

worker()
